k8s: Add SYS_MODULE capability, module paths

jrajahalme opened the pull request:
Add kernel module autoload capability, needed for IPv6 IPsec and
TPROXY in cases where the tables are not yet loaded when cilium-agent
starts.

Also add volume mounts for /lib/modules and /sbin/modprobe, as
'ip[6]tables' command depends on these when a kernel module is not
already loaded.

Fixes: #7453
Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/7581)
<!-- Reviewable:end -->

tgraf commented on examples/kubernetes/templates/v1/cilium-containerd-ds.yaml.sed:
```diff
@@ -131,6 +132,12 @@ spec:
         - mountPath: /tmp/cilium/config-map
           name: cilium-config-path
           readOnly: true
+        - mountPath: /lib/modules
```
Please add a comment to explain why it is needed

jrajahalme replied:
Added!

tgraf commented on examples/kubernetes/templates/v1/cilium-containerd-ds.yaml.sed:
```diff
@@ -131,6 +132,12 @@ spec:
+        - mountPath: /lib/modules
+          name: lib-modules
+          readOnly: true
+        - mountPath: /sbin/modprobe
```
Please add a comment to explain why it is needed

jrajahalme replied:
Added.

vadorovsky commented on examples/kubernetes/1.10/cilium-containerd-ds.yaml:
```diff
@@ -106,6 +106,7 @@ spec:
             add:
             - NET_ADMIN
+            - SYS_MODULE
           privileged: true
```
Maybe we should remove this line? As far as I know, `privileged: true` adds **all** capabilities that exist in the kernel. I think it's better to have only listed capabilities.

jrajahalme replied:
@tgraf I recall there being discussion that over some time we'd want to do exactly this (remove the `privileged: true` line, but that we are not there yet? I.e., we'll remove this in a future release after some more testing?

vadorovsky replied:
Alright, I'm fine with that. :+1: I was just curious and I thought that for some reason you need to define both `privileged: true` and list specific caps.

vadorovsky reviewed (changes requested):
One question.

jrajahalme committed:
k8s: Add SYS_MODULE capability, module paths

Add kernel module autoload capability, needed for IPv6 IPsec and
TPROXY in cases where the tables are not yet loaded when cilium-agent
starts.

Also add volume mounts for /lib/modules and /sbin/modprobe, as
'ip[6]tables' command depends on these when a kernel module is not
already loaded.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

vadorovsky reviewed (approved):
LGTM
